Scripts/Tracking/GameObjectComparer.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace VRTK.Core.Tracking
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+
+    /// <summary>
+    /// Compares a given <see cref="GameObject"/> with the <see cref="compareTo"/>.
+    /// </summary>
+    public class GameObjectComparer : MonoBehaviour
+    {
+        /// <summary>
+        /// Defines the event with the specified state.
+        /// </summary>
+        [Serializable]
+        public class UnityEvent : UnityEvent<GameObject>
+        {
+        }
+
+        /// <summary>
+        /// The <see cref="GameObject"/> to compare to.
+        /// </summary>
+        [Tooltip("The GameObject to compare to.")]
+        [SerializeField]
+        protected GameObject compareTo;
+
+        /// <summary>
+        /// Emitted when the given <see cref="GameObject"/> matches the current <see cref="compareTo"/> and emits the given <see cref="GameObject"/>.
+        /// </summary>
+        public UnityEvent Matched = new UnityEvent();
+        /// <summary>
+        /// Emitted when the given <see cref="GameObject"/> does not match the current <see cref="compareTo"/> and emits the given <see cref="GameObject"/>.
+        /// </summary>
+        public UnityEvent Mismatched = new UnityEvent();
+        /// <summary>
+        /// Emitted when the given <see cref="GameObject"/> matches the current <see cref="compareTo"/> and emits the <see cref="compareTo"/>.
+        /// </summary>
+        public UnityEvent MatchedAgainst = new UnityEvent();
+        /// <summary>
+        /// Emitted when the given <see cref="GameObject"/> does not match the current <see cref="compareTo"/> and emits the <see cref="compareTo"/>..
+        /// </summary>
+        public UnityEvent MismatchedAgainst = new UnityEvent();
+
+        /// <summary>
+        /// Sets the <see cref="compareTo"/> value.
+        /// </summary>
+        /// <param name="compareTo">The new value.</param>
+        public virtual void SetCompareTo(GameObject compareTo)
+        {
+            if (!IsValid())
+            {
+                return;
+            }
+
+            this.compareTo = compareTo;
+        }
+
+        /// <summary>
+        /// Clears the <see cref="compareTo"/> value.
+        /// </summary>
+        public virtual void ClearCompareTo()
+        {
+            compareTo = null;
+        }
+
+        /// <summary>
+        /// Compares the given <see cref="GameObject"/> with the <see cref="compareTo"/>.
+        /// </summary>
+        /// <param name="givenGameObject">The <see cref="GameObject"/> to compare against the <see cref="compareTo"/>.</param>
+        /// <returns><see langword="true"/> if the comparison is successful.</returns>
+        public virtual bool Compare(GameObject givenGameObject)
+        {
+            if (!IsValid())
+            {
+                return false;
+            }
+
+            bool result = (compareTo == givenGameObject);

```

Comment:
Instead of just having a `GameObject compareTo` I propose using an `RuleContainer rule` and implementing an `IRule` (inheriting from `BaseGameObjectRule`) that returns whether the passed GameObject is equal to a set one.

This won't change the intended usage here but results in a reusable component. I propose the type name changes to `RuleTester` or something.

Reply:
I'm wondering whether to just remove the GameObjectComparer commit completely, it's not used for anything right now

Reply:
True... Not too much work to recreate once we need it I think!